### PR TITLE
Listings

### LINKED ISSUE #6: Create New Listing
- [x] Run Listing Migration 
- [x] zip_code
- [x] produce_name (string)
- [x] produce_type (string)
- [x] description (string)
- [x] quantity (integer)
- [x] unit (string)
- [x] date_harvested (string)
- [x] status (string)
- [x] timestamps
- [x] build mutation createListing 
- [x] return id/errors

### PR DESCRIPTION
**Pull request by jdcarey128**
## Description

This PR adds the createListing mutation. 
Closes #6 

## Motivation and Context

This functionality is required to meet our MVP. 

## How Has This Been Tested?

GraphQL tests look at normal and edge case scenarios. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/graphql/mutations/create_listing.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    description "Create a listing record"
+    
+    argument :user_id, Integer, required: false
+    argument :zip_code, String, required: false
+    argument :produce_name, String, required: false
+    argument :produce_type, String, required: false
+    argument :description, String, required: false
+    argument :quantity, Integer, required: false
+    argument :unit, String, required: false
+    argument :date_harvested, String, required: false
```
Comment (dcoleman21) [inline app/graphql/mutations/create_listing.rb:12]: Ended up resorting to String here as a work around for DateTime.

File: app/graphql/types/listing_type.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module Types
+  class ListingType < Types::BaseObject
+    field :user, Types::UserType, null: false
```
Comment (dcoleman21) [inline app/graphql/types/listing_type.rb:3]: Added this to reference a User object in response

File: db/migrate/20210220152722_create_listings.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class CreateListings < ActiveRecord::Migration[5.2]
+  def change
+    create_table :listings do |t|
+      t.references :user, foreign_key: true
+      t.string :zip_code
+      t.string :produce_name
+      t.string :produce_type
+      t.string :description
+      t.numeric :quantity
```
Comment (dcoleman21) [inline db/migrate/20210220152722_create_listings.rb:9]: Ended up changing this to an integer

File: db/migrate/20210220171207_change_column_on_listings.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class ChangeColumnOnListings < ActiveRecord::Migration[5.2]
+  def change
+    change_column :listings, :quantity, :integer
```
Comment (dcoleman21) [inline db/migrate/20210220171207_change_column_on_listings.rb:3]: This changed the numeric to integer for quantity

File: spec/requests/create_listing_spec.rb [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    
+    post graphql_path, params: { query: query_string }
+
+    result = JSON.parse(response.body, symbolize_names: true)
+    error = result[:data][:createListing][:error]
+
+    expect(error[0]).to eq('User must exist')
+  end
+
+  it 'can return a user object type' do
```
Comment (dcoleman21) [inline spec/requests/create_listing_spec.rb:79]: Here is where we added `field :user, Types::UserType, null: false` in the type/listing_type.rb for this test

**Review by dcoleman21 — APPROVED** [review]
Great time figuring out the DateTime stuff with this mutation.

**Review by Callbritton — APPROVED** [review]
